src/download.ts

```diff
@@ -125,40 +179,60 @@ export function getHnsUrl(this: SkynetClient, domain: string, customOptions: any
   return addUrlQuery(url, query);
 }
 
-export function getHnsresUrl(this: SkynetClient, domain: string, customOptions: any = {}): string {
+/**
+ * Constructs the full URL for the resolver for the given HNS domain.
+ *
+ * @param this - SkynetClient
+ * @param domain - Handshake domain.
+ * @param [customOptions={}] - Additional settings that can optionally be set.
+ * @param [customOptions.endpointPath="/hnsres"] - The relative URL path of the portal endpoint to contact.
+ * @returns - The full URL for the resolver for the HNS domain.
+ */
+export function getHnsresUrl(this: SkynetClient, domain: string, customOptions?: BaseCustomOptions): string {
   const opts = { ...defaultResolveHnsOptions, ...this.customOptions, ...customOptions };
 
   domain = trimUriPrefix(domain, uriHandshakeResolverPrefix);
   return makeUrl(this.portalUrl, opts.endpointPath, domain);
 }
 
-export async function getMetadata(this: SkynetClient, skylink: string, customOptions: any = {}) {
+/**
+ * Gets only the metadata for the given skylink without the contents.
+ *
+ * @param this - SkynetClient
+ * @param skylinkStr - Skylink string. See `downloadFile`.
```

Comment:
Are you sure about this naming? I consistently see `skylink` everywhere else. Is this value somehow different?

Reply:
I want to distinguish somehow between a skylink and a string that could contain a skylink (e.g. a portal URL). Not sure what to name it, but I don't think `skylink` here was a great name because we have to extract the skylink from this string.

Reply:
I see. That makes sense. Maybe `skylinkUrl` can be more descriptive? It's just that `skylinkStr` did not immediately tell me that it's called that because there is some additional fluff in the string around the skylink itself.

Reply:
Sounds good to me.